src/rtp_receiver.rs

```diff
@@ -99,47 +101,68 @@ impl RtpReceiver {
         port: u16,
         should_close: Receiver<()>,
         packet_sender: Sender<Packet>,
-        decoder_params_sender: Sender<DecoderParameters>,
-    ) -> Result<()> {
+        decoder_params_sender: Sender<Result<DecoderParameters>>,
+    ) {
         let sdp_filepath = PathBuf::from(format!("/tmp/sdp_input_{}.sdp", port));
-        let mut file = File::create(&sdp_filepath)?;
-        file.write_all(
-            format!(
-                "\
-                    v=0\n\
-                    o=- 0 0 IN IP4 127.0.0.1\n\
-                    s=No Name\n\
-                    c=IN IP4 127.0.0.1\n\
-                    m=video {} RTP/AVP 96\n\
-                    a=rtpmap:96 H264/90000\n\
-                    a=fmtp:96 packetization-mode=1\n\
-                    a=rtcp-mux\n\
-                ",
-                port
-            )
-            .as_bytes(),
-        )?;
-        let input_ctx = input_with_dictionary_and_interrupt(
-            &sdp_filepath,
-            Dictionary::from_iter([("protocol_whitelist", "file,udp,rtp")]),
-            || should_close.try_recv().is_ok(),
-        )?;
-
-        let input = input_ctx
+
+        let result = File::create(&sdp_filepath)
```

Comment:
It's up to you, but I would extract all that code to separate function that returns a Result<> to avoid bunch of those `and_then` calls and matches on Results